File: Day1./Part1.java
mport java.io.File;
import java.io.IOException;
import java.util.Scanner;

public class Part1 {
	public static void main(String args[]) throws IOException {
        File file = new File("C:\\Users\\Andre\\eclipse-workspace\\Testing\\src\\advent1.txt");
		Scanner sc = new Scanner(file);
		int sum = 0;
		while(sc.hasNext()) {
			String line = sc.nextLine();
			int first = -1;
			int last = -1;
			int digit = -1;
			for(int i = 0; i < line.length(); i++) {
				if(Character.isDigit(line.charAt(i))) digit = Character.getNumericValue(line.charAt(i));
				
				if (digit != -1) {
					if(first == -1) first = digit;
					last = digit;
				}	
		}
		
		sum += first * 10 + last;
	}
	System.out.println(sum);
}
}

File: 2023/Day01/Part1.py
lines = open('test.txt')

sum = 0
curr = 0
for line in lines:
    for i, c in enumerate(line):
        if c.isdigit():
            curr += 10 * int(c)
            break

    for i, c in enumerate(reversed(line)):
        if c.isdigit():
            curr += int(c)
            break

    sum += curr
    curr = 0

print(sum)



